checks.d/consul.py

```diff
@@ -94,15 +95,14 @@ def _get_agent_datacenter(self, instance):
     def _is_instance_leader(self, instance):
         try:
             agent_url = self._get_agent_url(instance)
-            leader = self._last_known_leader or self._get_cluster_leader(instance)
```

Comment:
I may look at seeing if it can be preserved, but the one change here is that _last_known_leader isn't managed when in `self_leader_check` mode. Thus we need to look to see who is the leader each time this is called.